src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,40 +1,116 @@
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imageUrl, setImageUrl] = useState('');
+  const [imbdUrl, setImbdUrl] = useState('');
+  const [imbdId, setImbdId] = useState('');
+
+  const isFormFull = title && imageUrl && imbdUrl && imbdId;
+
+  const handleReset = () => {
+    setTitle('');
+    setDescription('');
+    setImageUrl('');
+    setImbdUrl('');
+    setImbdId('');
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!isFormFull) {
+      return;
+    }
+
+    const newMovieToPush: Movie = {
+      title: title,
+      description: description,
+      imgUrl: imageUrl,
+      imdbUrl: imbdUrl,
+      imdbId: imbdId,
+    };
+
+    onAdd(newMovieToPush);
+    setCount(current => current + 1);
+    handleReset();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form onSubmit={handleSubmit} className="NewMovie" key={count}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={event => setTitle(event)}
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        onChange={event => setDescription(event)}
+        required
+        name="description"
+        label="Description"
+        value={description}
+      />
 
-      <TextField name="imgUrl" label="Image URL" value="" />
+      <TextField
+        onChange={event => setImageUrl(event)}
+        required
+        name="imgUrl"
+        label="Image URL"
+        value={imageUrl}
+      />
 
-      <TextField name="imdbUrl" label="Imdb URL" value="" />
+      <TextField
+        onChange={event => setImbdUrl(event)}
+        required
+        name="imdbUrl"
+        label="Imdb URL"
+        value={imbdUrl}
+      />
 
-      <TextField name="imdbId" label="Imdb ID" value="" />
+      <TextField
+        onChange={event => setImbdId(event)}
+        required
+        name="imdbId"
+        label="Imdb ID"
+        value={imbdId}
+      />
 
       <div className="field is-grouped">
         <div className="control">
-          <button
-            type="submit"
-            data-cy="submit-button"
-            className="button is-link"
-          >
-            Add
-          </button>
+          {isFormFull ? (
+            <button
+              type="submit"
+              data-cy="submit-button"
+              className="button is-link"
+            >
+              Add
+            </button>
+          ) : (
+            <button
+              type="submit"
+              data-cy="submit-button"
+              className="button is-link"
+              disabled
+            >
+              Add
+            </button>
+          )}
```

Comment:
You can render just one button with a conditional disabled attribute, it simplifies the code, reduces repetition, and improves maintainability.
```suggestion
          <button
             type="submit"
             data-cy="submit-button"
             className="button is-link"
             disabled={!isFormFull}
           >
              Add
           </button>
```